frontera/core/models.py

```diff
@@ -111,6 +140,27 @@ def url(self):
         return self._url
 
     @property
+    def url_as_unicode(self):
+        """
+        Returns the URL as unicode.
+        """
+        return to_unicode(self.url, self.encoding)
+
+    @property
+    def url_as_bytes(self):
+        """
+        Returns the URL as bytes.
+        """
+        return to_bytes(self.url, self.encoding)
+
+    @property
+    def encoding(self):
```

Comment:
I haven't made the decision about whether the encoding needs to be stored. So right now the encoding is only stored in the Request/Response objects. I did this because we don't need to make this decision for single process mode's working.